### PR TITLE
doc: Condense README.md

### PR DESCRIPTION
**Pull request by mottefier**
I made a condensed Table of Contents format for the main github README.md in order to streamline the process for newer users. I found links to be repeated in some spots (src/test/README.md and /test in Automated Testing). The same information can be found elsewhere and in more context simply by visiting the link to the location needed by the user. The formatting is now similar to the documentation README.md

I added a separate file for Testing in order to not take up more space than necessary in the ToC.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,12 @@
-Bitcoin Core integration/staging tree
+**Bitcoin Core Integration/Staging Tree**
```
Comment (laanwj) [inline README.md:1]: There is no need to bold titles in markdown. The idea is to leave it up to the frontend how a certain style should look.

File: README.md [hunk 2]
```diff
@@ -17,66 +15,25 @@ with no central authority: managing transactions and issuing money are carried
...
--------
-
-Testing and code review is the bottleneck for development; we get more pull
-requests than we can review and test on short notice. Please be patient and help out by testing
-other people's pull requests, and remember this is a security-critical project where any mistake might cost people
-lots of money.
+For more information, read the original Bitcoin whitepaper.
 
-### Automated Testing
+[**Documents**](/doc) - Bitcoin Core Documentation
```
Comment (laanwj) [inline README.md:20]: Same for links, by the way.

File: doc/testing.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+**Testing**
```
Comment (MarcoFalke) [inline doc/testing.md:1]: I think this can be moved to `doc/README.md` instead

**Review by maflcko — COMMENTED** [review]
Seems fine to condense the "landing page" as long as all the info is linked to in another document.

**Comment by fanquake** [comment]
You need to [squash your commits](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits), and write a useful commit message, as well as address the linter failure:
```bash
This diff appears to have added new lines with trailing whitespace.
The following changes were suspected:

diff --git a/README.md b/README.md
@@ -25,2 +22 @@ License
+[Documents](/doc) - Bitcoin Core Documentation (see `doc/build-*.md` for instructions)
@@ -35,4 +29,2 @@ regularly from release branches to indicate new official, stable release version
+[Contributing Workflow](CONTRIBUTING.md) - Process/guidelines for contributing
@@ -53,8 +44 @@ lots of money.
+[Unit Tests](/src/test/README.md) - Developers are strongly encouraged to write unit tests for new code and to submit new unit tests for old code. Unit tests can be compiled and run (assuming they weren't disabled in configure) with: `make check`.
^---- failure generated from test/lint/lint-whitespace.sh
```

**Comment by MarcoFalke** [comment]
Closing for now due to inactivity

**Pull request closed without merging**